components/nexus-blobstore-file/src/main/java/org/sonatype/nexus/blobstore/file/MapdbBlobMetadataStore.java

```diff
@@ -64,6 +64,8 @@
 
   private TxMaker database;
 
+  private final SizeCache sizeCache = new SizeCache();
```

Comment:
I'd lean towards putting the `size` and `numRecords` fields here and avoid the cost of the extra lookup + method calls.